torchvision/transforms/functional_pil.py

```diff
@@ -261,8 +261,7 @@ def pad(img, padding, fill=0, padding_mode="constant"):
         raise ValueError("Padding mode should be either constant, edge, reflect or symmetric")
 
     if padding_mode == "constant":
-        if isinstance(fill, numbers.Number):
-            fill = (fill,) * len(img.getbands())
+        fill = _parse_fill(fill, img, "2.3.0")["fillcolor"]
```

Comment:
I've traced back the `min_version` to https://github.com/python-pillow/Pillow/commit/20b9d9774a2fb3f3be3a445b470905f9737ac362, but you could probably go even further back. Still, I think its sufficient because I think a lot of the functionalities we are using require a higher version be default